src/rebootstrap-raft/getmachineid_test.go

```diff
@@ -0,0 +1,88 @@
+package main
+
+import (
+	"io/fs"
+	"os"
+	"testing"
+)
+
+type mockDirEntry struct {
+	name  string
+	isDir bool
+	info  fs.FileInfo
+}
+
+func (m mockDirEntry) Name() string {
+	return m.name
+}
+
+func (m mockDirEntry) IsDir() bool {
+	return m.isDir
+}
+
+func (m mockDirEntry) Type() fs.FileMode {
+	if m.isDir {
+		return fs.ModeDir
+	}
+	return fs.ModeType
+}
+
+func (m mockDirEntry) Info() (fs.FileInfo, error) {
+	return nil, nil
+}
+
+func TestMissingAgentDirectory(t *testing.T) {
+	command := &rebootstrapCommand{}
+	_, err := command.getMachineID()
+	if err.Error() != "reading directory \"/var/lib/juju/agents\": open /var/lib/juju/agents: no such file or directory" {
+		t.Errorf("Expected an error for missing agent directory, got %s", err)
+	}
+}
+
+func TestGetID(t *testing.T) {
+	readDir = func(string) ([]os.DirEntry, error) {
+		return []os.DirEntry{&mockDirEntry{"machine-0", true, nil}}, nil
+	}
+	command := &rebootstrapCommand{}
+	machineID, err := command.getMachineID()
+	if err != nil {
+		t.Errorf("Expected no error, got %s", err)
+	}
+	if machineID != "0" {
+		t.Errorf("Expected machine ID to be '0', got %s", machineID)
+	}
+}
+
+func TestMultipleMachineIDs(t *testing.T) {
+	readDir = func(string) ([]os.DirEntry, error) {
+		return []os.DirEntry{
+			&mockDirEntry{"machine-0", true, nil},
```

Comment:
If we do go with this structure, they would be easier to understand as:
```suggestion
			&mockDirEntry{name: "machine-0", isDir: true, info: nil},
```

Any time you can use the struct names, especially for booleans, it makes it a lot more obvious to someone coming into the test.